codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,489 @@
+            return false;
+        }
+
+        int localHours = -1;
+        int localMinutes = -1;
+        int localSeconds = -1;
+        int currentPartNumber = 0;
+        int currentPartValue = 0;
+
+        for (int i = tokenStart; i < tokenEnd; i++) {
+            char c = txt.charAt(i);
+            if (isDigit(c)) {
+                currentPartValue = currentPartValue * 10 + getNumericalValue(c);
+            } else if (c == ':') {
+                if (currentPartValue == 0) {
+                    // invalid :: (nothing in between)
+                    return false;
+
+                } else if (currentPartNumber == 0) {
+                    // flushing hours
+                    localHours = currentPartValue;
+                    currentPartValue = 0;
+                    currentPartNumber++;
+
+                } else if (currentPartNumber == 1) {
+                    // flushing minutes
+                    localMinutes = currentPartValue;
+                    currentPartValue = 0;
+                    currentPartNumber++;
+
+                } else if (currentPartNumber == 2) {
+                    // invalid, too many :
+                    return false;
+                }
+            } else {
+                // invalid char
+                return false;
+            }
+        }
+
+        if (currentPartValue > 0) {
+            // pending seconds
+            localSeconds = currentPartValue;
+        }
+
+        if (localHours >= 0 && localMinutes >= 0 && localSeconds >= 0) {
+            hours = localHours;
+            minutes = localMinutes;
+            seconds = localSeconds;
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean tryParseDayOfMonth(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len != 1 && len != 2) {
+            return false;
+        }
+
+        int localDayOfMonth = 0;
+        for (int i = tokenStart; i < tokenEnd; i++) {
+            char c = txt.charAt(i);
+            if (isDigit(c)) {
+                localDayOfMonth = localDayOfMonth * 10 + getNumericalValue(c);
+            } else {
+                // invalid
+                return false;
+            }
+        }
+
+        if (localDayOfMonth > 0) {
+            dayOfMonth = localDayOfMonth;
+            return true;
+        }
+
+        return false;
+    }
+
+    private static boolean regionMatchesIgnoreCase(CharSequence txt, int offset, String other) {
+        if (txt instanceof String) {
+            return ((String) txt).regionMatches(true, offset, other, 0, other.length());
+        } else {
+            int end = Math.min(offset + other.length(), txt.length());
+            return txt.subSequence(offset, end).toString().equalsIgnoreCase(other);
+        }
+    }
+
+    private boolean tryParseMonth(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len == 3) {
```

Comment:
Sorry, I don't get you. If the CharSequence is a String, I don't have to copy anything. Otherwise, I just have to copy 3 chars instead of the whole header.